### PR TITLE
Postinglist.size to correspond to the number of unique docID

### PR DESCRIPTION
**Pull request by C-E-OOI**

### CODE REVIEW

File: index.py [hunk 1]
```diff
@@ -105,8 +105,12 @@ def build(self):
             if i == 0 or term != tokens_list[i-1][0]:
                 # new term
                 self.dictionary[term] = PostingList()
-            # insert into appropriate PostingList
-            self.dictionary[term].insert(curr_tuple[0], curr_tuple[1], curr_tuple[2])
+            # Insert into appropriate PostingList
+            # if same term and docID, do not increment PL.size
+            if term == tokens_list[i-1][0] and curr_tuple[0] == tokens_list[i-1][1][0]:
```
Comment (hcwong) [inline index.py]: Might have an error here if `i` is 0 cos `tokens_list[-1]`

Comment (hcwong) [reply index.py]: Might want to check if `i==0`

File: index.py [hunk 2]
```diff
@@ -315,12 +319,13 @@ def get_size(self):
...
     # Insert with var byte encoding
-    def insert(self, doc_id, field, positions):
+    def insert(self, doc_id, field, positions, new_doc_id=True):
         next_id = self.size
         new_posting = Posting(next_id, doc_id, field, positions)
         new_posting.var_byte_encoding()
         self.postings.append(new_posting)
-        self.size += 1
+        if new_doc_id:
+            self.size += 1
```
Comment (hcwong) [inline index.py:328]: Let's rename `self.size` to `self.unique/distinct_docids` for clarity

Comment (hcwong) [reply index.py:328]: Though some code on the search side relies on the `size` attribute of PostingList, so those might have to be changed to `len(posting_list.postings)` before merging

File: index.py (hunk 1 above)
Comment (C-E-OOI) [reply index.py]: ok

**Pull request merged**